evaluate_library_access_from_output_tsv/create_stratefied_sample_of_dois.R

```diff
@@ -0,0 +1,71 @@
+#### Load dependencies ####
+
+# Load magrittr pipe
+`%>%` = dplyr::`%>%`
+
+#### Settings ####
+
+lzma_compressed_library_access_tsv_location <- file.path(
+  'data', 'library_coverage_xml_and_fulltext_indicators.tsv.xz'
+)
+
+original_dataset_with_oa_color_column_location <- paste0(
+  'https://github.com/greenelab/scihub/raw/',
+  '4172526ac7433357b31790578ad6f59948b6db26/data/',
+  'state-of-oa-dois.tsv.xz'
+)
+
+sample_size_per_cell <- 100  # This will be for each cell, multiplied by 
+# 2 full_text_indicator status
+
+output_tsv_location <- file.path(
+  'evaluate_library_access_from_output_tsv',
+  'manual-doi-checks.tsv'
+)
+
+randomizer_seed_to_set <- 3  # Ensure that random sampling will always return the same result.
+
+#### Read and merge datasets ####
+
+source(file.path(
+  'evaluate_library_access_from_output_tsv',
+  'read_and_merge_library_access_datasets.R'
+))
+
+# The function below will return a merged dataset, but will *also* add the
+# following datasets to our global environment:
+  # original_dataset_with_oa_color_column
+  # lzma_compressed_library_access_tsv
+
+merged_datasets <- read_and_merge_library_access_datasets(
+  lzma_compressed_library_access_tsv_location,
+  original_dataset_with_oa_color_column_location
+)
+
+# Convert variables to factors:
+merged_datasets$oadoi_color <- factor(merged_datasets$oadoi_color)
+merged_datasets$full_text_indicator <- factor(
+  merged_datasets$full_text_indicator
+)
+
+#### Create stratefied sample within full_text_access and oadoi_color ####
+
+set.seed(randomizer_seed_to_set)
+stratefied_sample <- merged_datasets %>%
+  dplyr::group_by(full_text_indicator) %>%
+  dplyr::sample_n(sample_size_per_cell)
+
+#### Add columns to fill in manually to the stratefied sample dataframe ####
+
+colnames(stratefied_sample)[3] <- 'full_text_indicator_automated'
```

Comment:
`dplyr::mutate` to create new columns

Reply:
Implemented in 0d9724a.

Reply:
Avoid referring to columns by index rather than name:

```
colnames(stratefied_sample)[3] <- 'full_text_indicator_automated'
```

Are you intending to do a `dplyr::rename` here?

Then everything can be chained together so you're only assigning stratefied_sample a single time.

Reply:
Ah shit, readr probably can only read gz URLs:

> Remote gz files can also be automatically downloaded and decompressed.

https://github.com/tidyverse/readr/issues/163
https://github.com/tidyverse/readr/commit/22d60358ad22379f81de06a7b5af400fc6de0270

That's really annoying. Do you need to read this file anymore here?

Reply:
Re:  `dplyr::rename`, implemented in 8e6e9f3d.

Reply:
Opened https://github.com/tidyverse/readr/issues/764. But don't wait for this to be addressed (likely will take months to years before a release with the feature comes out).

Reply:
I don't need to read the second file *per se*, since we're no longer stratifying within `oadoi_color`. Is your preference that I remove it? The current (non-`readr`) code does work as-is, if we want to retain it.

Reply:
> Is your preference that I remove it?

Yes! Less is more!

Reply:
Implemented at approx. aac5bf5ef0.